### ISSUE TITLE
[Refactor][plugins] Reduce the file size of plugin `so` files

### ISSUE THREAD

**Issue by klesh** [apache/incubator-devlake#4213]
## What and why to refactor
Currently, each plugin takes up to `33mb` for its `so` binary, which adds up quickly and makes our docker image huge.

![image](https://user-images.githubusercontent.com/61080/212598273-f20d3351-016e-43d2-9051-53a9de9f9fab.png)

## Describe the solution you'd like
Refactor our plugin and compilation script to reduce the binary file size.
Reference: https://sonyarouje.com/2021/04/25/go-plugin-file-size-optimization/
Simply put, we should define a set of interfaces for our `helpers`, and pass the pointer for plugins to utilize

**Comment by keon94** [comment]
@klesh would this also reduce compilation times for the plugins?

**Comment by klesh** [comment]
However, this is a huge refactor. The `helpers` must be well define as set of interfaces and plain structs then pass into plugins as pointers
